File: app/src/main/java/com/focus/mob/ObjectifPrincipalActivity.java
package com.focus.mob;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;
import androidx.appcompat.app.AppCompatActivity;

public class ObjectifPrincipalActivity extends AppCompatActivity {

    private View[] cards;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_objectif_principal);

        ImageButton btnBack = findViewById(R.id.btn_back);
        Button btnContinuer = findViewById(R.id.btn_continuer);

        cards = new View[]{
                findViewById(R.id.card_study),
                findViewById(R.id.card_work),
                findViewById(R.id.card_read),
                findViewById(R.id.card_create),
                findViewById(R.id.card_other)
        };

        for (View card : cards) {
            card.setOnClickListener(v -> selectCard(v));
        }

        btnBack.setOnClickListener(v -> finish());

        btnContinuer.setOnClickListener(v -> {
            Intent intent = new Intent(ObjectifPrincipalActivity.this, NiveauDistractionActivity.class);
            startActivity(intent);
        });
    }

    private void selectCard(View selectedCard) {
        for (View card : cards) {
            card.setSelected(card == selectedCard);
        }
    }
}

File: app/src/main/java/com/focus/mob/ui/onboarding/ObjectifPrincipalActivity.kt
package com.focus.mob.ui.onboarding
import com.focus.mob.R

import com.focus.mob.ui.auth.*
import com.focus.mob.ui.main.*
import com.focus.mob.ui.onboarding.*
import com.focus.mob.ui.session.*
import com.focus.mob.utils.*

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.focus.mob.databinding.ActivityObjectifPrincipalBinding

class ObjectifPrincipalActivity : AppCompatActivity() {

    private lateinit var binding: ActivityObjectifPrincipalBinding
    private lateinit var cards: Array<View>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityObjectifPrincipalBinding.inflate(layoutInflater)
        setContentView(binding.root)

        cards = arrayOf(
            binding.cardStudy, binding.cardWork,
            binding.cardRead, binding.cardCreate, binding.cardOther
        )

        cards.forEach { card -> card.setOnClickListener { selectCard(it) } }

        binding.btnBack.setOnClickListener { finish() }

        binding.btnContinuer.setOnClickListener {
            startActivity(Intent(this, NiveauDistractionActivity::class.java))
            fadeTransition()
        }
    }

    private fun selectCard(selectedCard: View) {
        cards.forEach { it.isSelected = (it == selectedCard) }
    }
}
